tests/NavItem.test.ts

```diff
@@ -0,0 +1,49 @@
+/// <reference lib="dom" />
+
+import { describe, expect, beforeEach, afterEach, it, mock } from 'bun:test';
+
+import NavItem from '../src/NavItem';
+
+describe.skip('NavItem', () => {
+
+  let navItem: NavItem;
+
+  beforeEach(() => {
+    navItem = document.createElement('nav-item');
+    document.body.appendChild(navItem);
+
+    if (navItem == null) return;
+  });
+
+  afterEach(() => {
+    document.body.removeChild(navItem);
+  });
+
+  it('should initialize correctly', () => {
+    expect(navItem).toBeTruthy();
+  });
+
+  it('should not render if not connected to DOM', () => {
+    const test = document.createElement('nav-item');
+    const href = test.querySelector('a');
+    expect(href).not.toBeTruthy();
+  });
+
+  it('should invoke click event when the user clicks on it', () => {
+    const cb = mock(e => e.preventDefault());
+
+    navItem.click();
```

Comment:
You are not registering the callback **cb** to the navItem anywhere. It will never be called.